db/migrate/20250303121736_create_challenges.rb

```diff
@@ -0,0 +1,11 @@
+class CreateChallenges < ActiveRecord::Migration[8.0]
+  def change
+    create_table :challenges do |t|
+      t.string :name, null: false
+      t.references :owner, null: false, foreign_key: { to_table: :users }
+      t.date :start_date, null: false
+      t.date :end_date, null: false
```

Comment:
Start and end should be represented as `time`. Then I propose to keep rails convention and rename it to:

```suggestion
      t.time :starts_at, null: false
      t.time :ends_at, null: false
```